plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/SyncPartitionMetadataProcedure.java

```diff
@@ -67,6 +68,8 @@
         ADD, DROP, FULL
     }
 
+    private static final int BATCH_GET_PARTITIONS_BY_NAMES_MAX_PAGE_SIZE = 1000;
```

Comment:
Playing with the batch size, it seems that higher batches makes it faster.
With 1000 it took 3.59m
With 5000 it took 2.73m
I think that `BATCH_GET_PARTITIONS_BY_NAMES_MAX_PAGE_SIZE` should be configurable.